src/Bridge/Laravel/DbToolsServiceProvider.php

```diff
@@ -0,0 +1,270 @@
+            return new LaravelDatabaseSessionRegistry($app->make('db'));
+        });
+
+        $this->app->singleton(ConfigurationRegistry::class, function (Application $app) {
+            /** @var Repository $config */
+            $config = $app->make('config');
+
+            $defaultConfig = new Configuration(
+                backupBinary: $config->get('db-tools.backup_binary', '/usr/bin/pg_dump'),
+                backupExcludedTables: $config->get('db-tools.backup_excluded_tables'),
+                backupExpirationAge: $config->get('db-tools.backup_expiration_age', '3 months ago'),
+                backupOptions: $config->get('db-tools.backup_options', '-Z 5 --lock-wait-timeout=120'),
+                backupTimeout: $config->get('db-tools.backup_timeout', 600),
+                restoreBinary: $config->get('db-tools.restore_binary', '/usr/bin/pg_restore'),
+                restoreOptions: $config->get('db-tools.restore_options', '-j 2 --clean --if-exists --disable-triggers'),
+                restoreTimeout: $config->get('db-tools.restore_timeout', 1800),
+                storageDirectory: $config->get('db-tools.storage_directory', $app->storagePath('db_tools')),
+                storageFilenameStrategy: $config->get('db-tools.storage_filename_strategy', 'default'),
+            );
+
+            $connectionConfigs = [];
+            foreach ($config->get('db-tools.connections', []) as $name => $data) {
+                $connectionConfigs[$name] = new Configuration(
+                    backupBinary: $data['backup_binary'] ?? null,
+                    backupExcludedTables: $data['backup_excluded_tables'] ?? null,
+                    backupExpirationAge: $data['backup_expiration_age'] ?? null,
+                    backupOptions: $data['backup_options'] ?? null,
+                    backupTimeout: $data['backup_timeout'] ?? null,
+                    restoreBinary: $data['restore_binary'] ?? null,
+                    restoreOptions: $data['restore_options'] ?? null,
+                    restoreTimeout: $data['restore_timeout'] ?? null,
+                    parent: $defaultConfig,
+                    storageDirectory: $data['storage_directory'] ?? null,
+                    storageFilenameStrategy: $data['storage_filename_strategy'] ?? null,
+                );
+            }
+
+            return new ConfigurationRegistry(
+                $defaultConfig,
+                $connectionConfigs,
+                $config->get('database.default')
+            );
+        });
+
+        $this->app->singleton(Storage::class, function (Application $app) {
+            /** @var ConfigurationRegistry $configRegistry */
+            $configRegistry = $app->make(ConfigurationRegistry::class);
+
+            $connections = $configRegistry->getConnectionConfigAll();
+            if (empty($connections)) {
+                $connections[$configRegistry->getDefaultConnection()] = $configRegistry->getDefaultConfig();
```

Comment:
You don't need this if statement, the ConfigurationRegistry object does it for you when requested.

Reply:
Ok! I will remove this `if` :ok_hand:

Reply:
Wait, `ConfigurationRegistry` class does not handle this case. Method `getConnectionConfigAll()` may not be complete.

Reply:
I'm not sure, but I think that getConnectionConfigAll is not required for most commands

Reply:
Now I'm sure, the getConnectionConfigAll() method only serves the purpose of knowing connections in the standalone bootstrap code, but in laravel, it's handled in the specific database session component instead.

Reply:
I looked at the `Storage` class more precisely and I realized that I actually can reduce this part of code :+1: